sdk/azcore/etag.go

```diff
@@ -0,0 +1,80 @@
+// +build go1.13
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azcore
+
+import (
+	"strings"
+)
+
+// ETag is a property used for optimistic concurrency during updates
+// ETag is a validator based on https://tools.ietf.org/html/rfc7232#section-2.3.2
+type ETag struct {
+	value *string
+}
+
+// NewETag creates a new ETag struct
+func NewETag(value string) *ETag {
+	return &ETag{value: &value}
+}
+
+// ComparisonType specifies what type of comparison to use, Strong or Weak
+type ComparisonType string
+
+const (
```

Comment:
I suggest we get rid of these constants and have WeakEquals and StrongEquals methods instead.